app/components/dns-record/form.tsx

```diff
@@ -0,0 +1,63 @@
+import { AddIcon, InfoIcon } from '@chakra-ui/icons';
+import { Button, Input, Select, Textarea, Tooltip, VStack } from '@chakra-ui/react';
+import { Form } from '@remix-run/react';
+import FormField from './form-field';
+
+interface FormParams {
+  typeError?: string; // Error for 'Type' field
+}
+
+export default function DomainForm({ typeError }: FormParams) {
+  return (
+    <Form className="domain-form" method="post">
+      <VStack maxW="xl" spacing="2">
+        <FormField label="Domain Name" isRequired={true}>
+          <Input name="name" />
```

Comment:
In a follow-up, let's use an [`<InputRightAddon>`](https://chakra-ui.com/docs/components/input#left-and-right-addons) to show the right-hand parts of the domain (i.e., we only want `foo` in `foo.username.domain.com`, so show `{username}.domain.com` on the right.)